qt/5.15.2: fails to build on macOS 12.0.1 Monterrey

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: qt/5.15.2
  * Operating System+version: macOS 12.0.1
  * Compiler+version: clang 13
  * Conan version: 1.42.0
  * Python version: 3.9.7


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Macos
os_build=Macos
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13.0
build_type=Release
arch=x86_64
arch_build=x86_64
[options]
[conf]
[build_requires]
[env]
```

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
compiling ~/.conan/data/qt/5.15.2/_/_/source/qt5/qtbase/src/plugins/platforms/cocoa/qcocoasessionmanager.cpp
In file included from ~/.conan/data/qt/5.15.2/_/_/source/qt5/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.mm:40:
~/.conan/data/qt/5.15.2/_/_/source/qt5/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h:54:32: error: unknown type name 'CGColorSpaceRef'; did you mean 'QColorSpace'?
    void setColorSpace(QCFType<CGColorSpaceRef> colorSpace);
                               ^~~~~~~~~~~~~~~
                               QColorSpace
~/conan/data/qt/5.15.2/_/_/source/qt5/qtbase/include/QtCore/../../src/corelib/kernel/qmetatype.h:2090:1: note: 'QColorSpace' declared here
QT_FOR_EACH_STATIC_GUI_CLASS(QT_FORWARD_DECLARE_STATIC_TYPES_ITER)
^
~/.conan/data/qt/5.15.2/_/_/source/qt5/qtbase/include/QtCore/../../src/corelib/kernel/qmetatype.h:178:24: note: expanded from macro 'QT_FOR_EACH_STATIC_GUI_CLASS'
    F(QColorSpace, 87, QColorSpace) \
                       ^
~/.conan/data/qt/5.15.2/_/_/source/qt5/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.mm:90:32: error: out-of-line definition of 'setColorSpace' does not match any declaration in 'QIOSurfaceGraphicsBuffer'
void QIOSurfaceGraphicsBuffer::setColorSpace(QCFType<CGColorSpaceRef> colorSpace)
                               ^~~~~~~~~~~~~
~/.conan/data/qt/5.15.2/_/_/source/qt5/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h:54:24: note: type of 1st parameter of member declaration does not match definition ('QCFType<QColorSpace>' vs 'QCFType<CGColorSpaceRef>')
    void setColorSpace(QCFType<CGColorSpaceRef> colorSpace);
                       ^
2 errors generated.

```

</details>

--- a/recipes/qt/5.x.x/conandata.yml
+++ b/recipes/qt/5.x.x/conandata.yml
@@ -24,3 +24,5 @@ patches:
       base_path: "qt5/qtdeclarative"
     - patch_file: "patches/f830b86.diff"
       base_path: "qt5/qtwebengine/src/3rdparty"
+    - patch_file: "patches/dece6f5.diff"
+      base_path: "qt5/qtbase"

--- a/recipes/qt/5.x.x/patches/dece6f5.diff
+++ b/recipes/qt/5.x.x/patches/dece6f5.diff
@@ -1,0 +1,14 @@
+diff --git a/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h b/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
+index e070ba977d..35a62f59e3 100644
+--- a/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
++++ b/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h
+@@ -43,6 +43,9 @@
+ #include <qpa/qplatformgraphicsbuffer.h>
+ #include <private/qcore_mac_p.h>
+ 
++
++#include <CoreGraphics/CGColorSpace.h>
++
+ QT_BEGIN_NAMESPACE
+ 
+ class QIOSurfaceGraphicsBuffer : public QPlatformGraphicsBuffer

